protocol/protocol.re

```diff
@@ -60,26 +60,47 @@ let apply_side_chain = (state: t, operation) => {
   let included_operations = Set.add(operation, state.included_operations);
 
   let {source, amount, ticket, _} = operation;
-  let.ok (ledger, result) =
-    switch (operation.kind) {
-    | Transaction({destination}) =>
-      let.ok ledger =
-        Ledger.transfer(~source, ~destination, amount, ticket, state.ledger);
-      Ok((ledger, `Transaction));
-    | Withdraw({owner}) =>
-      let.ok (ledger, handle) =
-        Ledger.withdraw(
-          ~source,
-          ~destination=owner,
-          amount,
-          ticket,
-          state.ledger,
-        );
-      // TODO: publish the handle somewhere
-      Ok((ledger, `Withdraw(handle)));
-    };
-
-  Ok(({...state, ledger, included_operations}, result));
+  switch (operation.kind) {
+  | Transaction({destination}) =>
+    let.ok ledger =
+      Ledger.transfer(~source, ~destination, amount, ticket, state.ledger);
+    Ok(({...state, ledger, included_operations}, `Transaction));
+  | Withdraw({owner}) =>
+    let.ok (ledger, handle) =
+      Ledger.withdraw(
+        ~source,
+        ~destination=owner,
+        amount,
+        ticket,
+        state.ledger,
+      );
+    // TODO: publish the handle somewhere
+    Ok(({...state, ledger, included_operations}, `Withdraw(handle)));
+  | Add_validator(validator, timestamp) =>
+    let validators = Validators.add(validator, state.validators);
+    let last_seen_membership_change_timestamp = timestamp;
+    Ok((
+      {
+        ...state,
+        last_seen_membership_change_timestamp,
+        validators,
+        included_operations,
+      },
+      `Add_validator(validator),
+    ));
+  | Remove_validator(validator, timestamp) =>
+    let validators = Validators.remove(validator, state.validators);
+    let last_seen_membership_change_timestamp = timestamp;
+    Ok((
+      {
+        ...state,
+        validators,
+        last_seen_membership_change_timestamp,
+        included_operations,
+      },
+      `Remove_validator(validator),
+    ));
+  };
```

Comment:
This logic is duplicated from Add_validator

Reply:
abstracted into a single function. For now, Add/Remove are separate operations.